siliconcompiler/apps/sc_remote.py

```diff
@@ -0,0 +1,113 @@
+    parser.add_argument('-credentials', nargs='?', default=default_credentials_file())
+
+    args = parser.parse_args()
+
+    # Sanity checks.
+    if (args.reconnect and (args.cancel or args.delete)):
+        print('Error: -reconnect is mutually exclusive to -cancel and -delete')
+        return 1
+    elif (args.cancel and (args.reconnect or args.delete)):
+        print('Error: -cancel is mutually exclusive to -reconnect and -delete')
+        return 1
+    elif (args.delete and (args.reconnect or args.cancel)):
+        print('Error: -delete is mutually exclusive to -reconnect and -cancel')
+        return 1
+    elif ((args.reconnect or args.cancel or args.delete) and not args.jobid):
+        print('Error: -jobid is required for -reconnect, -cancel, and -delete')
+        return 1
+    elif (args.reconnect and not args.cfg):
+        print("Error: -cfg is required for -reconnect. Recommended value is "
+              "the post-import manifest in the job's original build directory.")
+        return 1
+
+    # Read in credentials from file, if specified and available.
+    # Otherwise, use the default server address.
+    if os.path.isfile(args.credentials):
+        with open(args.credentials, 'r') as cfgf:
+            try:
+                remote_cfg = json.loads(cfgf.read())
+            except json.JSONDecodeError:
+                print('Error reading remote configuration file: invalid JSON')
+                return 1
+    else:
+        # TODO: I think this default is stored somewhere - client.py? _metadata.py?
+        remote_cfg = {'address': 'https://server.siliconcompiler.com',
+                      'port': '443'}
+
+    # Main logic.
+    # If no job-related options are specified, fetch and report basic info.
+    # Create temporary Chip object and check on the server.
+    chip = Chip('server_test')
+    chip.status['remote_cfg'] = remote_cfg
+    chip.status['jobhash'] = args.jobid
+    remote_ping(chip)
+
+    # If the -cancel flag is specified, cancel the job.
+    if (args.jobid and args.cancel):
+        cancel_job(chip)
+
+    # If the -delete flag is specified, delete the job.
+    elif (args.jobid and args.delete):
+        delete_job(chip)
+
+    # If the -reconnect flag is specified, re-enter the client flow
+    # in its "check_progress/ until job is done" loop.
+    elif (args.jobid and args.reconnect):
+        # TODO: Will require optional '-cfg' argument, we can't reconnect
+        # without the job's manifest. If we update the server to return the
+        # design name in the 'check_progress/' response, we could instead accept
+        # optional '-builddir' and '-jobname' arguments, then get the design
+        # and node names from a call to 'check_progress/'.
+        # Also, total runtime value will be incorrect; maybe we can have the
+        # server return the job's "created_at" time in the check_progress/ response.
+        remote_run_loop(chip, time.monotonic())
```

Comment:
Lets not print the wrong time.
Not really sure what you mean by optionally -cfg?

Reply:
We can't "reconnect" to a remote job without knowing its design name, build directory, flowgraph nodes, etc. So until we add a local mapping of job IDs to build directories, we need `-cfg` in addition to `-jobid` to resume a remote job.

Reply:
This won't actually impact the output - we don't print the total time in the progress check, just the runtime for each node which is reported by the server.